File: javascript/specs/blackbox/continuations.spec.js
const enolib = require('../..');

const input = `
Field: [value]
\\ [spaced_line_continuation]
| [direct_line_continuation]

Fieldset:
Entry = [value]
\\ [spaced_line_continuation]
| [direct_line_continuation]

List:
- [value]
\\ [spaced_line_continuation]
| [direct_line_continuation]

Empty field:
\\ [spaced_line_continuation]
| [direct_line_continuation]

Fieldset with empty entry:
Empty entry =
\\ [spaced_line_continuation]
| [direct_line_continuation]

List with empty item:
-
\\ [spaced_line_continuation]
| [direct_line_continuation]
`.trim();

describe('Blackbox test', () => {
  describe('Continuations', () => {
    it('performs as expected', () => {
      expect(enolib.parse(input).raw()).toMatchSnapshot();
    });
  });
});

File: ruby/spec/blackbox/continuations.spec.rb
# frozen_string_literal: true

input = <<~DOC.strip
Field: Value
\\ Spaced continuation
| Direct continuation

Field:
Attribute = Value
\\ Spaced continuation
| Direct continuation

List:
- Value
\\ Spaced continuation
| Direct continuation

Field:
\\ Spaced continuation
| Direct continuation

Field with attribute:
Attribute =
\\ Spaced continuation
| Direct continuation

Field with item:
-
\\ Spaced continuation
| Direct continuation
DOC

describe 'Blackbox test' do
  describe 'Continuations' do
    it 'performs as expected' do
      # TODO: Replace .raw with custom spec helper
      expect(Enolib.parse(input).raw).to match_snapshot
    end
  end
end
